Source/AwsCommonRuntimeKit/crt/ByteBuffer.swift

```diff
@@ -290,129 +74,19 @@ extension ByteBuffer: IStreamable {
-            } else if read == 0 {
-                // EOF
-                break
-            }
-            allocate(read)
-            putByte(buffer.pointee)
-        }
-    }
-
-    /// initialize a new `ByteBuffer` instance from a file path in the format of a `String`
-    public convenience init(filePath: String) throws {
-        let url = URL(fileURLWithPath: filePath)
-
-        let data = try Data(contentsOf: url)
-
-        self.init(data: data)
-    }
-
-    /// Coverts the array of bytes to a `Data` object
-    ///
-    /// - Returns: `Data`
-    public func toData() -> Data {
-        return Data(bytes: array, count: array.count)
-    }
-
-    /// Creates a new file at the `filePath` given and writes the data to it returning a `FileHandle` to it.
-    ///
-    /// - Parameters:
-    ///   - filePath:  The path at which you would like a new file to be created at and written to.
-    /// - Returns: `FileHandle`
-    public func toFile(filePath: String) -> FileHandle? {
-        let fileManager = FileManager.default
-        if fileManager.createFile(atPath: filePath, contents: self.toData()) {
-            return FileHandle(forReadingAtPath: filePath)
-        } else {
-            return nil
-        }
-    }
-
-    /// Converts the array of bytes to `Foundation.InputStream`
-    ///
-    /// - Returns: `InputStream`
-    public func toStream() -> InputStream {
-        return InputStream(data: self.toData())
-    }
-
-    func toAwsByteBuf(allocator: Allocator = defaultAllocator) -> aws_byte_buf {
-        let count = self.array.count
-        return self.array.withUnsafeMutableBufferPointer { pointer in
-            return aws_byte_buf(len: count,
-                                buffer: pointer.baseAddress,
-                                capacity: capacity,
-                                allocator: allocator.rawValue)
-        }
-    }
-}
-
-public extension ByteBuffer {
-    func sha256(allocator: Allocator = defaultAllocator, truncate: Int = 0) -> ByteBuffer {
-        var byteCursor = aws_byte_cursor_from_array(self.array, self.array.count)
-
-        var bytes = [UInt8](repeating: 0, count: Int(AWS_SHA256_LEN))
-        let result: ByteBuffer = bytes.withUnsafeMutableBufferPointer { pointer in
-            var buffer = aws_byte_buf(len: 0, buffer: pointer.baseAddress, capacity: Int(AWS_SHA256_LEN), allocator: allocator.rawValue)
-            aws_sha256_compute(allocator.rawValue, &byteCursor, &buffer, truncate)
-            return buffer.toByteBuffer()
-        }
-        return result
-    }
-
-    func base64EncodedSha256(allocator: Allocator = defaultAllocator, truncate: Int = 0) -> String {
-        return sha256(allocator: allocator, truncate: truncate).toData().base64EncodedString()
-    }
-
-    func encodeToHexString() -> String {
-        var hexString = ""
-        for byte in array {
-            hexString += String(format: "%02x", UInt8(byte))
-        }
-
-        return hexString
-    }
```

Comment:
This is being used in the SDK. It is a simple helper function that belongs in the SDK and not here.